### ISSUE TITLE
[Slack] Add messages to Later with reminder

### ISSUE THREAD

**Issue by gabriel-r** [raycast/extensions#15424]
### Extension

https://www.raycast.com/mommertf/slack

### Description

Create a command to provide the following:
1. Send a message "Get bread" (user input) to user "slackbot", or to "me"
2. Set a default reminder of 1h

_UPDATE 2025-04-11: It's now possible to add a [standalone personal reminder](https://slack.com/help/articles/208423427-Set-a-reminder#h_01J8FMHE6AC5ZTT81XQ1ZTEWSZ) to the Later list in Slack, without the need to attach a reminder to a message. This would be even better here._

Some nice to have features:

* The default reminder duration should be customizable. 
* The reminder could be parsed from the message content, i.e. "Get bread tomorrow" or "Get bread in 1h"

### Who will benefit from this feature?

People using Slack as a task manager, with reminders on messages. This method is more flexible and useful than the built-in Reminders feature in Slack.

**Comment by RobErskine** [comment]
I could potentially take this on if other people feel like it would be useful? We don't have a way to _send_ messages via Raycast but maybe we could?

I think this could just be a more generic "Send Message" command. Could use the <Form> layout to pull all users/channels for who you want to send it to, the message itself and then a time for which you want to send (now, 1hr, or set specific datetime). Could use the [send or schedule message API](https://api.slack.com/messaging/sending)?

**Comment by RobErskine** [comment]
It's been a little bit but now that the holidays are over I've started to work on this if you want to assign this over to me..

**Comment by RobErskine** [comment]
[PR is up for this here](https://github.com/raycast/extensions/pull/16580) 😎 

![Image](https://github.com/user-attachments/assets/ebd280ed-a741-4bd0-94b2-e9ede72cf60d)

**Comment by RobErskine** [comment]
hey @gabriel-r, there unfortunately hasn't been any movement on testing [my PR](https://github.com/raycast/extensions/pull/16580) feature to add in the `Send Message` command. However there was [a new extension](https://www.raycast.com/koh-sh/slack-templated-message) published to the store earlier today which looks to do this with templated messages baked in. Check this out and see if this satisfies your use case for now: https://www.raycast.com/koh-sh/slack-templated-message

**Comment by gabriel-r** [comment]
Thanks for much Rob, much appreciated 🙇 
I'll try the other extension, looks promising.

**Comment by alexandrelissowski** [comment]
@RobErskine I don't know if your PR does it, but I would love to have direct access to all messages I mark as `Later` from Raycast. 

Is it possible to add a command to list all `Later` messages? 

On each `Later` message, we could: 
- jump to the message in Slack 
- Mark it as `Complete` 
- Remind it in 1h 

If my feedback is not at the right place, I could create a new feedback issue to give more details. 

Thanks

**Comment by RobErskine** [comment]
@alexandrelissowski thanks for the feedback.

My PR is all about sending messages directly from Raycast, either immediately (right now) or at a scheduled date (in 1 hr, tomorrow at 9am, etc). The latter is the "Later" feature I think you are referring to, which is something different. 

I think what you are talking about is a way to view all of the messages that you have marked to review "Later":

![Mark message for review later](https://github.com/user-attachments/assets/00d64d7b-f358-430d-94d1-5bb399846f9b).

I think you are referring to a new command where you could see all messages that you have "Saved for Later". Unfortunately this PR doesn't address that. 

However I think what you are looking for is something similar to the "Unread Messages" command, but instead is a "Saved for Later Messages" command. Is that right? If so, feel free to create a new issue request and mention me in it. I'd be happy to take a look at that.

**Comment by RobErskine** [comment]
@gabriel-r this command was approved and published today! 

![Image](https://github.com/user-attachments/assets/5aa64ae5-dc07-4aed-a2b3-d228260059a3)

Check Raycast to update your extensions and it should be in there. 

@raycastbot I think this can be moved to Close

**Comment by gabriel-r** [comment]
To be clear, I need a way to quickly add items to the Later list in Slack
This is not that, but it's a useful addition.
Thanks!

**Comment by RobErskine** [comment]
Oh, sorry @gabriel-r, not sure where the wires got crossed. I didn't understand the "Later" requirement. Based on that feedback, I think this PR accomplishes half of your original ask?

> ✅ Send a message "Get bread" (user input) to user "slackbot", or to "me"
❓ Set a default reminder of 1h

When you said "Reminder" it made me think of scheduling a message to be sent, but I think what you were referring to was adding the sent message to the "Later" feature of Slack, with a reminder time to be added to it. Is that correct?

As brought up in #18160 the "Later" API is not available yet, so the second part of your request may need to be put on ice while we wait for that to come out.

May need to add the `[status: blocked by 3rd party API](https://github.com/raycast/extensions/issues?q=state%3Aopen%20label%3A%22status%3A%20blocked%20by%203rd%20party%20API%22)` to this.
